frappe/core/doctype/scheduled_job_type/scheduled_job_type.py

```diff
@@ -106,7 +106,7 @@ def get_next_execution(self):
 			self.cron_format = CRON_MAP[self.frequency]
 
 		return croniter(
-			self.cron_format, get_datetime(self.last_execution or datetime(2000, 1, 1))
+			self.cron_format, get_datetime(self.last_execution or now_datetime())
```

Comment:
This is probably a bad idea. 

Imagine a new job that's added to run at 12am. Now because of scheduler tick scheduler runs at 11:58 and then at 12:02, the job will never run once.

Reply:
We can subtract `scheduler_interval` (typically 4 minutes) from the now_datetime. That should help the cold start.

Reply:
Or even better solution. Set last_execution to current `now_datetime` when document is inserted, so we never have to handle this special handling code.

Reply:
We are not comparing dates exactly, it is <=, so it will fire in next interval.

Reply:
> We are not comparing dates exactly, it is <=, so it will fire in next interval.


It won't. Run the test I added with old commits.